frontend/src/pages/Status.tsx

```diff
@@ -49,31 +52,38 @@ export function hasFinished(status?: NodePhase): boolean {
     case NodePhase.SUCCEEDED: // Fall through
     case NodePhase.FAILED: // Fall through
     case NodePhase.ERROR: // Fall through
-    case NodePhase.SKIPPED:
+    case NodePhase.SKIPPED: // Fall through
+    case NodePhase.TERMINATED:
       return true;
     case NodePhase.PENDING: // Fall through
     case NodePhase.RUNNING: // Fall through
+    case NodePhase.TERMINATING: // Fall through
     case NodePhase.UNKNOWN:
       return false;
     default:
       return false;
   }
 }
 
-export function statusToBgColor(status?: NodePhase): string {
+export function statusToBgColor(status?: NodePhase, errorMessage?: string): string {
```

Comment:
nit: Would it be cleaner if the status is reconciled with the error message before being passed here, rather than have this function do the reconciliation?

Reply:
Neither is great because what we're doing isn't great. I think this  approach has a slightly lower chance of mistakes (e.g. failing to provide error / failing to call disambiguation function) being made because the extra param serves as a reminder (i think i can give it a better name), but I agree that it's not ideal and am willing to have the disambiguation function called outside  of these functions